cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -174,11 +174,10 @@ def transform_cis_soc_data(df: DataFrame, join_on_columns: List[str]) -> DataFra
     """
     transform and process cis soc data
     """
+
     df = df.withColumn(
         "standard_occupational_classification_code",
-        F.when(F.substring(F.col("standard_occupational_classification_code"), 1, 2) == "un", "uncodeable").otherwise(
-            F.col("standard_occupational_classification_code")
-        ),
+        F.regexp_replace(F.col("standard_occupational_classification_code"), r"[a-zA-Z].*|^\s*$", "uncodeable"),
```

Comment:
Additional new requirement - please create a flag column to indicate which records had an invalid SOC code and were edited to "uncodeable"

Reply:
added